File: C#/Loon2MonoGame/LoonMonoGame-Lib/loon/Save.cs
using System.Collections.Generic;

namespace loon
{

    public interface Batch
    {

        void SetItem(string key, string data);

        void RemoveItem(string key);

        void Commit();
    }

    public interface Save
    {


         void SetItem(string key, string data);

         void RemoveItem(string key);

         string GetItem(string key);

         Batch StartBatch();

         IEnumerable<string> Keys();

         bool IsPersisted();
    }
}

File: Java/Loon-Neo/src/loon/Save.java
/**
 * Copyright 2008 - 2015 The Loon Game Engine Authors
 * 
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not
 * use this file except in compliance with the License. You may obtain a copy of
 * the License at
 * 
 * http://www.apache.org/licenses/LICENSE-2.0
 * 
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations under
 * the License.
 * 
 * @project loon
 * @author cping
 * @email：javachenpeng@yahoo.com
 * @version 0.5
 */
package loon;

public interface Save {

	interface Batch {

		void setItem(String key, String data);

		void removeItem(String key);

		void commit();
	}

	public void setItem(String key, String data);

	public void removeItem(String key);

	public String getItem(String key);

	public Batch startBatch();

	public Iterable<String> keys();

	public boolean isPersisted();
}
